docs/explanation/security.md

```diff
@@ -0,0 +1,78 @@
+(security)=
+
+# Security
+
+## Cryptographic technology
+
+The only case where Ops uses cryptography is for sending trace data, when a certificate authority provider has been integrated with the charm. This is done using HTTPS, using either TLS 1.2 or 1.3, using the implementation provided by the standard library of the Python that is executing the charm. See [tracing security](#ops_tracing_security) for more details.
+
+There is no use of hashing or digital signatures.
+
+## Inter-process communication
+
+Ops communicates with Juju by reading environment variables and running processes in the charm environment (hook commands).
+
+> See also:
+> - {external+juju:ref}`Juju | Hook <hook>`
+> - {external+juju:ref}`Juju |Hook command <list-of-hook-commands>`
+
+## Hardening
+
+Hardening a charm that uses Ops is done in the same way as any other charm: no extra hardening steps are required as a result of using Ops.
+
+> See also: {external+juju:ref}`Juju | Harden your deployment <harden-your-deployment>`
+
+## Charm unit databases
+
+Ops stores state in a sqlite3 database named `.unit-state.db`. This database includes [](ops.StoredState) objects and the defer notice queue.
+
+The state database is in the charm directory, which is set by Juju in the `JUJU_CHARM_DIR` environment variable and typically looks like `/var/lib/juju/agents/unit-my-unit-0/charm`. For a machine charm, this directory is on the unit's machine. For a Kubernetes charm, this directory is in the charm container.
+
+Ops sets the permissions for the state database to allow only reading and writing by the user running the charm. You shouldn't try to edit this file or change its permissions.
+
+Ops buffers tracing data in a sqlite3 database named `.tracing-data.db` in the same directory as the state database. Trace data is stored in this database only if the `tracing` extra is installed. When `tracing` is installed, Ops buffers tracing data even if tracing has not been enabled through an integration with a trace receiver (this allows collecting traces prior to the integration).
+
+For example, the permissions of the databases are:
+
+```text
+-rw-r--r--  1 root root  32K Jul 13 23:48 .tracing-data.db
+-rw-------  1 root root  20K Jul 13 23:48 .unit-state.db
```

Comment:
Not a critique of this doc, but: I guess it's intended that the tracing db is readable by group and all, whereas the unit state db is only r/w by root?